src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -766,19 +766,48 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
     }
 
     void Schedule::handleGRUPNET(HandlerContext& handlerContext) {
-        for (const auto& record : handlerContext.keyword) {
-            const auto& groupName = record.getItem("NAME").getTrimmedString(0);
-
-            if (!this->snapshots.back().groups.has(groupName))
-                addGroup(groupName , handlerContext.currentStep);
 
-            int table = record.getItem("VFP_TABLE").get< int >(0);
+        auto network = this->snapshots.back().network.get();
+        std::vector<Network::Node> nodes;
+        for (const auto& record : handlerContext.keyword) {
+            const std::string& groupNamePattern = record.getItem<ParserKeywords::GRUPNET::NAME>().getTrimmedString(0);
+            const auto group_names = this->groupNames(groupNamePattern);
+            if (group_names.empty())
+                this->invalidNamePattern(groupNamePattern, handlerContext);
+            const auto& pressure_item = record.getItem<ParserKeywords::GRUPNET::TERMINAL_PRESSURE>();
+            const int vfp_table = record.getItem<ParserKeywords::GRUPNET::VFP_TABLE>().get<int>(0);
 
-            auto new_group = this->snapshots.back().groups.get( groupName );
-            if (new_group.updateNetVFPTable(table))
-                this->snapshots.back().groups.update( std::move(new_group) );
+            for (const auto& group_name : group_names) {
+                auto& group = this->snapshots.back().groups.get(group_name);
+                group.updateNetVFPTable(vfp_table);
+                const auto& parent_name = group.parent();
+                if (parent_name != "")
+                {
+                    const std::string& downtree_node = group_name;
+                    const std::string& uptree_node = parent_name;
+                    if (vfp_table == 0 && network.has_node(downtree_node) && network.has_node(uptree_node)) {
+                        network.drop_branch(uptree_node, downtree_node);
+                    } else {
+                        const auto alq_eq = Network::Branch::AlqEqfromString(record.getItem<ParserKeywords::GRUPNET::ALQ_SURFACE_DENSITY>().get<std::string>(0));
+                        if (alq_eq == Network::Branch::AlqEQ::ALQ_INPUT) {
+                            double alq_value = record.getItem<ParserKeywords::GRUPNET::ALQ>().get<double>(0);
+                            network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_value));
+                        } else {
+                            network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_eq));
+                        }
+                    }
+                }
+                Network::Node node { group_name };
+                if (pressure_item.hasValue(0) && (pressure_item.get<double>(0) > 0))
```

Comment:
because when the `pressure_item` is specified with a valid pressure (non-defaulted and non-negative value (`>=` might be better if sticking to the text)), it will be a fixed-pressure node. Then the VFP table **must** be defaulted (we should throw if NOT) here and there is no connection to its parent group. 

if the `pressure_item` is defaulted or specified with a negative value, it is not a fixed-node pressure. Then we can handle the rest of the items in the record. 

As a result, I am suggesting re-organize the code a little bit to have the above logic in an if .. els e structure. 

``` c++
const bool valid_node_pressure = pressure_item.hasValue(0) && (pressure_item.get<double>(0) >= 0));

if (valid_node_pressure) {
    // adding a fixed pressure node
} else {
    // handling the rest of the record to handle the branch
}
```

We might also need to drop a node if a non-fixed pressure node has defaulted VFP table number, not totally clear to me how we should do it yet or what it implies yet (@atgeirr might be able to comment on this). You can just also throw for that situation for now.

Reply:
I followed your suggestions and added the two throws you suggested.